### ISSUE TITLE
Support list placement for text

### ISSUE THREAD

**Issue by gravitystorm** [mapbox/carto#238]
Since mapnik 2.1 you can provide a list of placement options for the textsymbolizer. This is an alternative to the "simple" placement type.

```
<TextSymbolizer face-name="DejaVu Sans Book" size="16" placement="point" dy="8" fill="blue" placement-type="list">[name]
    <Placement size="10" dy="-8" fill="red"/><!-- Reduces text size and changes offset -->
    <Placement fill="green">[abbreviated_name]</Placement> <!-- size="10", dy="-8", fill="green", shorter text -->
    <Placement fill="orange" dy="8">[nr]</Placement> <!-- size="10", dy="8", fill="orange", shortest text -->
</TextSymbolizer>
```

(from https://github.com/mapnik/mapnik/wiki/TextSymbolizer )

I don't have a particularly clear idea what the carto syntax would look like for this!

See also https://github.com/mapnik/mapnik-reference/issues/45 , since it's not yet in mapnik-reference either.

**Comment by kocio-pl** [comment]
Is there a chance to add this support into the Carto? It allows to fine tune the style.

**Comment by nebulon42** [comment]
Definitely. The main shortcoming for CartoCSS here at the moment are developers who are willing to submit patches. However, for this specifically it will be first necessary to find a proper way to represent this in CartoCSS (see #347). There is a unfinished PR that needs to be picked up and probably rewritten from scratch.

Current plans are to release 1.0 in early 2018 without this and tackle this afterwards.
